spec/unit/functions/hiera_spec.rb

```diff
@@ -121,5 +122,12 @@
       hiera_include.expects(:call_function_with_scope).with(scope,'include', %w[key foo])
       hiera_include.call(scope, 'foo') { |k| ['key', k] }
     end
+
+    it 'should do nothing when called during environment compilation' do
+      scope = create_test_scope_for_environment('production')
+      hiera_include = scope.compiler.loaders.puppet_system_loader.load(:function, 'hiera_include')
```

Comment:
This does not test the full code path from a call in a manifest. Suggest using `scope.call_function('hiera_include', ['foo']) instead.
Also, since this assumes that 'foo' is a key in hiera data that would return something other than nil, maybe check that the expected debug message is printed to ensure it hits the overridden function (the proc) instead of just returning nil for "not found"